src/check/constrain/generate/operation.rs

```diff
@@ -116,24 +123,24 @@ pub fn gen_op(
 
         Node::BOneCmpl { expr } => {
             let left = Expected::try_from((expr, &env.var_mappings))?;
-            constr.add("binary compliment", &left, &Expected::new(expr.pos, &ExpressionAny));
+            constr.add("binary compliment", &left, &Expected::new(expr.pos, &Type { name: Name::any() }));
```

Comment:
What if we have a function for type so we can have `Expect::Any`. 
Implement `Any` interface for `Expect`?